velox/exec/Window.cpp

```diff
@@ -350,54 +496,19 @@ void Window::callApplyForPartitionRows(
     rawPeerEnds[j] = peerEndRow_ - 1 - firstPartitionRow;
   }
 
-  auto updateFrameBounds = [&](vector_size_t* rawFrameBounds,
-                               core::WindowNode::BoundType boundType,
-                               core::WindowNode::WindowType type,
-                               bool isStartBound) -> void {
-    switch (boundType) {
-      case core::WindowNode::BoundType::kUnboundedPreceding:
-        std::memset(rawFrameBounds, 0, numRows * sizeof(vector_size_t));
-        break;
-      case core::WindowNode::BoundType::kUnboundedFollowing:
-        std::fill_n(
-            rawFrameBounds, numRows, lastPartitionRow - firstPartitionRow);
-        break;
-      case core::WindowNode::BoundType::kCurrentRow: {
-        if (type == core::WindowNode::WindowType::kRange) {
-          vector_size_t* rawPeerBuffer =
-              isStartBound ? rawPeerStarts : rawPeerEnds;
-          std::copy(rawPeerBuffer, rawPeerBuffer + numRows, rawFrameBounds);
-        } else {
-          // Fills the frameBound buffer with increasing value of row indices
-          // (corresponding to CURRENT ROW) from the startRow of the current
-          // output buffer. The startRow has to be adjusted relative to the
-          // partition start row.
-          std::iota(
-              rawFrameBounds,
-              rawFrameBounds + numRows,
-              startRow - firstPartitionRow);
-        }
-        break;
-      }
-      case core::WindowNode::BoundType::kPreceding:
-      case core::WindowNode::BoundType::kFollowing:
-        VELOX_NYI("Not supported");
-      default:
-        VELOX_USER_FAIL("Invalid frame bound type");
-    }
-  };
-
   for (auto i = 0; i < numFuncs; i++) {
-    updateFrameBounds(
-        rawFrameStartBuffers[i],
-        windowFrames_[i].startType,
-        windowFrames_[i].type,
-        true);
-    updateFrameBounds(
-        rawFrameEndBuffers[i],
-        windowFrames_[i].endType,
-        windowFrames_[i].type,
-        false);
+    updateFrameBounds(i, true, startRow, numRows);
```

Comment:
Why did you change the parameters here ? The original ones seem more efficient and easier to understand.

Reply:
Since the function ```updateFrameBounds``` now belongs to class Window, the rest of the parameters can be derived from the ```functionIndex```. So only the minimal set of parameters required are used as arguments to simplify the function.
Also, since```updateFrameBounds``` is only called twice per window function, the computation of these parameters would not be expensive. Please let me know if these params should be passed as arguments instead.

Reply:
Thanks for the suggestion, updated as discussed. CI is also now fixed with this change.